enhancements/security/security-profiles-operator.md

```diff
@@ -0,0 +1,636 @@
+---
+title: security-profiles-operator
+authors:
+  - "@jaormx"
+  - "@jhrozek"
+  - "@saschagrunert"
+reviewers:
+  - "@ashcrow"
+  - "@mrunalp"
+  - "@sttts"
+  - "@mrunalp"
+  - "@shawn-hurley"
+  - "@dmesser"
+approvers:
+  - TBD
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced|informational
+see-also:
+  - "/enhancements/this-other-neat-thing.md"
+replaces:
+  - [selinux-operator proposal](https://github.com/openshift/enhancements/pull/327)
+superseded-by:
+  - "/enhancements/our-past-effort.md"
+---
+
+# Security Profiles Operator Integration in OpenShift
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Let's integrate the [Security Profiles Operator](https://github.com/kubernetes-sigs/security-profiles-operator)
+as part of the OpenShift platform in order to allow folks to:
+
+* Easily install security profiles to secure their workloads
+  (both SELinux and Seccomp)
+
+* Manage the lifecycle of the aforementioned profiles (make sure
+  that they get attached to the appropriate container as upgrades
+  happen)
+
+* Enable users to automatically generate profiles for their workloads
+
+* Provide predefined profiles for existing default workloads in OpenShift
+
+
+## Motivation
+
+OpenShift already has several security controls enabled by default. Amongst them
+are the following:
+
+* SELinux to restrict what resources can containers interact with
+
+* Seccomp to restrict the system calls that containers are allowed to do
+
+* Security Context Constraints to control what security knobs and handles are
+  service accounts able to tweak in their workloads.
+
+The default SELinux and Seccomp profiles work great for restricted workloads.
+However, once the workload needs a little more permissions, setting these
+security settings becomes non-trivial. In face of this, developers often
+respond by turning off these controls and running their workloads as privileged
+containers.
+
+The [Security Profiles Operator (SPO)](https://github.com/kubernetes-sigs/security-profiles-operator)
+is an upstream project aiming to ease installation and development of security
+profiles in Kubernetes. By using these profiles, developers would have a more
+granular way of setting permissions for their workloads, hopefully eliminating
+the need to run privileged containers
+
+### Goals
+
+* Integrate the Security Profiles Operator into OpenShift by having it
```

Comment:
Can we list the motivation for including this by default in the core?
We are working hard to reduce our platform footprint, so the bar for including something by default is very high.

From a security posture, it makes sense that we provide good defaults out of the box that work for most of our customers. For the ones that do want to further lock down or loosen their security, they could install
this operator using OLM.

Reply:
> Can we list the motivation for including this by default in the core?

The main motivation was to not require that installing workloads that currently require to run as privileged to install an add-on, but rather provide a way to ship policies along with the operators.

> We are working hard to reduce our platform footprint, so the bar for including something by default is very high.
> 

Nod.

> From a security posture, it makes sense that we provide good defaults out of the box that work for most of our customers. For the ones that do want to further lock down or loosen their security, they could install
> this operator using OLM.

Perhaps the way to go might be a dependency on another operator, as said elsewhere. But ultimately, we should be locking down everything, including the core platform operators. And at that point, isn't it nicer to have something like SPO that lets you treat the security policies as first-class kubernetes objects?

Reply:
> From a security posture, it makes sense that we provide good defaults out of the box that work for most of our customers. For the ones that do want to further lock down or loosen their security, they could install
this operator using OLM.

If we want to provide good defaults, reducing the number of privileged containers we ship on the core OCP would be a good step (right now it's around 150+ privileged containers ). We believe that having this operator be installed by default would help us get there, as it would more easily enable our devs to start locking down their workload without needing to go through extra hoops.